File: src/main/java/org/example/tree/InvertBinaryTree.java
package org.example.tree;

public class InvertBinaryTree {

    public TreeNode invertTree(TreeNode root) {
        if (root == null || (root.left == null && root.right == null)) return root;

        TreeNode left = root.left;
        TreeNode right = root.right;
        root.left = right;
        root.right = left;
        invertTree(left);
        invertTree(right);

        return root;
    }
}

File: src/main/java/org/example/tree/InvertBinaryTree.py
from typing import Optional

from org.example.tree.TreeNode import TreeNode


class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root or (not root.left and not root.right): return root

        root.left, root.right = root.right, root.left
        self.invertTree(root.left)
        self.invertTree(root.right)

        return root
